r_sort.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python
+from merge_sort import timed_func
+
+
+@timed_func
+def r_sort(sequence):
+    digit = 1
+    while True:
+        bins = [[] for i in range(10)]
+        for x in sequence:
+            # the expression in bins returns the number in digit place
+            bins[x % 10 ** digit // 10 ** (digit -1)].append(x)
+        sequence = [number for x in bins for number in x]
+        digit += 1
+        if len(sequence) == len(bins[0]):
+            return sequence
```

Comment:
This is a really interesting way of figuring out how to stop the sorting passes.  The problem I have with it is that it entails doing one extra pass through the sorting that doesn't really need to happen.  Can you figure out before you start how many passes through the sorting step you'll need to make?